dataland-frontend/src/utils/DataOwnerUtils.ts

```diff
@@ -0,0 +1,34 @@
+import { assertDefined } from "@/utils/TypeScriptUtils";
+import type Keycloak from "keycloak-js";
+import { ApiClientProvider } from "@/services/ApiClients";
+import { type AxiosError } from "axios";
+import { waitForAndReturnResolvedKeycloakPromise } from "@/utils/KeycloakUtils";
+
+/**
+ * Check if a user is data owner of a company
+ * @param companyId the dataland companyId of the company for which ownership should be checked
+ * @param keycloakPromiseGetter the getter-function which returns a Keycloak-Promise
+ * @returns a promise, which resolves to a boolean
+ */
+export async function isUserDataOwnerForCompany(
+  companyId: string,
+  keycloakPromiseGetter?: () => Promise<Keycloak>,
+): Promise<boolean> {
+  const uuidRegexExp = /^[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}$/i;
+  const isCompanyIdValid = uuidRegexExp.test(companyId);
```

Comment:
Wieso hast du den format-Check hier mit reingeommen?

Ich glaube der Endpunkt selbst beschwert sich bereits, wenn man ihm eine nicht-valide UUID gibt.
Daher ist es hier redundant denke ich.

Reply:
Ja der Endpunkt beschwert sich schon. Aber ohne den Check fliegt mir einer der FrameworkViewPage Tests um die Ohren, weil Cypress den 500 Fehler nicht handlen kann.
Und ehrlicherweise macht es auch nicht wirklcih Sinn, mit einem schon falschen Input ans Backend zu gehen. Daher hatte ich den Check hier eingebaut....

Reply:
Ich habe es ncohmal ohne den Check versucht, doch dann ist der Test wieder fehlgeschlagen. Daher habe ich den Check erst einmal drin behalten

Reply:
Ist zwar unschön, aber macht unser Produkt nicht "kaputt".  Daher schließe ich das jetzt.

Aber es fühlt sich als Leser wirklich sehr random an,  hier "versteckt" plötzlich eine UUID-Format-Validierung zu finden.

Vllt sollten wir das in künftigen Tickets zentral checken.

Reply:
Bin ich dafür. Das geht in die gleiche Richtung wie der Check für das Anzeigen der SummaryPanel Cards. Ich glaube das könnte alles mit einem ähnlichen Check abgefangen werden.
Im Prinzip müsste man nur sicherstellen, dass bei der Anfrage einer Company ans Backend, kein Fehler zurück kommt. Und wenn man keinen Fehler erhalten hat, dann wird der Titel der Firma angezeigt, die Panel werden angezeigt und man macht den Check auf Data Ownership.
Wenn man einen Fehler bekommenhat, dann zeigt man  nur an, diese Firma gibt es nicht, keine Panels und macht auch keinen Data Owner Check.